src/python/WMCore/MicroService/MSPileup/DataStructs/MSPileupObj.py

```diff
@@ -0,0 +1,139 @@
+    "containerFraction": real number with the container fraction to be distributed (TBFD)
+    "replicationGrouping": string with a constant value (DATASET or ALL, to be in sync with Rucio)
+    "activatedOn": int, seconds since epoch in GMT timezone
+    "deactivatedOn": int, seconds since epoch in GMT timezone
+    "active": boolean,
+    "pileupSizeGB": integer, current size of the pileup in GB (integer)
+    "rulesList: list of strings (rules) used to lock the pileup id
+}
+
+The data flow should be done via list of objects, e.g.
+[<pileup object 1>, <pileup object 2>, ..., <pileup object n>]
+"""
+
+# system modules
+import time
+import json
+
+# WMCore modules
+from WMCore.MicroService.Tools.Common import getMSLogger
+
+
+def gmtimeSeconds():
+    """
+    Return GMT time in seconds
+    """
+    return int(time.mktime(time.gmtime()))
+
+
+class MSPileupObj(object):
+    """
+    MSPileupObj defines MSPileup data stucture
+    """
+    def __init__(self, pdict, verbose=None, logger=None):
+        self.logger = getMSLogger(verbose, logger)
+
+        self.pileupName = pdict.get('pileupName', '')
+        self.pileupType = pdict.get('pileupType', 'classic')
+        self.insertTime = pdict.get('insertTime', gmtimeSeconds())
+        self.lastUpdateTime = pdict.get('lastUpdateTime', gmtimeSeconds())
+        self.expectedRSEs = pdict.get('expectedRSEs', [])
+        self.fullReplicas = pdict.get('fullReplicas', 0)
+        self.campaigns = pdict.get('campaigns', [])
+        self.containerFraction = pdict.get('containerFraction', 0.0)
+        self.replicationGrouping = pdict.get('replicationGrouping', "")
+        self.activatedOn = pdict.get('activatedOn', gmtimeSeconds())
+        self.deactivatedOn = pdict.get('deactivatedOn', gmtimeSeconds())
+        self.active = pdict.get('active', False)
+        self.pileupSize = pdict.get('pileupSize', 0)
+        self.ruleList = pdict.get('ruleList', [])
+
+        self.data = {
+            'pileupName': self.pileupName,
+            'pileupType': self.pileupType,
+            'insertTime': self.insertTime,
+            'lastUpdateTime': self.lastUpdateTime,
+            'expectedRSEs': self.expectedRSEs,
+            'fullReplicas': self.fullReplicas,
+            'campaigns': self.campaigns,
+            'containerFraction': self.containerFraction,
+            'replicationGrouping': self.replicationGrouping,
+            'activatedOn': self.activatedOn,
+            'deactivatedOn': self.deactivatedOn,
+            'active': self.active,
+            'pileupSize': self.pileupSize,
+            'ruleList': self.ruleList}
+        if not self.validate(self.data):
+            msg = 'MSPileup input is invalid'
+            raise Exception(msg)
+
+    def __str__(self):
+        """
+        Return human readable representation of pileup data
+        """
+        return json.dumps(self.data, indent=4)
+
+    def validate(self, pdict):
```

Comment:
Valentin, we should implement value specific validation as well. A non-exhaustive list would be:
1) validate `pileupName` value to pass this Lexicon `dataset` method: https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Lexicon.py#L208
2) `pileupType` in `('classic', 'premix')`
3) `expectedRSEs` and `currentRSEs`. My suggestion here is to make an specific method for this validation. My idea is that this `MSPileupObj` class accepts a list of RSEs as input parameter and uses it to validate RSE values (this way we do not need to add Rucio service dependency into a data structure module)
4) `containerFraction` between 0.0 and 1.0
5) `replicationGrouping` in `('DATASET', 'ALL')`

Please create the relevant unit tests.

Reply:
added initial implementation but need to provide unit tests for all values (will come back to it once I'll get time next week).

Reply:
done